### PR TITLE
Add telemetry for platform versions being used

### PR DESCRIPTION
**Pull request by qianz2**

### CODE REVIEW

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 1]
```diff
@@ -65,12 +65,20 @@ public string GenerateBashScript(RunScriptGeneratorContext ctx)
                     var detectionResult = platform.Detect(ctx);
                     if (detectionResult != null)
                     {
-                        _logger.LogDebug($"Detected platform '{detectionResult.Platform}' with version '{detectionResult.PlatformVersion}'.");
+                        _logger.LogDebug($"Detected platform '{detectionResult.Platform}' with version " +
```
Comment (arroyc) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs:68]: if we already had these .. why no telelmetry is there?

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 2]
```diff
@@ -65,12 +65,20 @@ public string GenerateBashScript(RunScriptGeneratorContext ctx)
                     var detectionResult = platform.Detect(ctx);
                     if (detectionResult != null)
                     {
-                        _logger.LogDebug($"Detected platform '{detectionResult.Platform}' with version '{detectionResult.PlatformVersion}'.");
+                        _logger.LogDebug($"Detected platform '{detectionResult.Platform}' with version " + 
+                        "'{detectionResult.PlatformVersion}'.");
                         if (string.IsNullOrEmpty(detectionResult.PlatformVersion))
                         {
-                            throw new UnsupportedVersionException($"Couldn't detect a version for platform '{detectionResult.Platform}' in the repo.");
+                            throw new UnsupportedVersionException($"Couldn't detect a version for platform " +
```
Comment (arroyc) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs:72]: add one more log for not able to detect a version for a platform

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs [hunk 3]
```diff
@@ -65,12 +65,20 @@ public string GenerateBashScript(RunScriptGeneratorContext ctx)
...
-                        _logger.LogDebug($"Detected platform '{detectionResult.Platform}' with version '{detectionResult.PlatformVersion}'.");
+                        _logger.LogDebug($"Detected platform '{detectionResult.Platform}' with version " + 
+                        "'{detectionResult.PlatformVersion}'.");
                         if (string.IsNullOrEmpty(detectionResult.PlatformVersion))
                         {
-                            throw new UnsupportedVersionException($"Couldn't detect a version for platform '{detectionResult.Platform}' in the repo.");
+                            throw new UnsupportedVersionException($"Couldn't detect a version for platform " + 
+                            "'{detectionResult.Platform}' in the repo.");
                         }
+                        var buildEventProps = new Dictionary<string, string>()
```
Comment (arroyc) [inline src/BuildScriptGenerator/DefaultRunScriptGenerator.cs:75]: nit: buildeventtelemetryprop

File: src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs [hunk 4]
```diff
@@ -225,7 +225,13 @@ internal class DefaultBuildScriptGenerator : IBuildScriptGenerator
                     _logger.LogDebug(
                         "Platform {platformName} with version {platformVersion} was used.",
                         platform.Name,
-                        detectorResult);
+                        detectorResult.PlatformVersion);
+                    var buildEventProps = new Dictionary<string, string>()
```
Comment (arroyc) [inline src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs:229]: nit:buildeventtelemetryprop

File: src/BuildScriptGenerator/DefaultRunScriptGenerator.cs (hunk 1 above)
Comment (qianz2) [reply src/BuildScriptGenerator/DefaultRunScriptGenerator.cs:68]: I just found it in the telemetry under "GetBuildSnippets" custom events, probably have missed it. Though not found the telemtry for the runScript platform versions. Will send the screenshot to the team.

**Pull request closed without merging**